compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -1033,6 +1033,15 @@ fn variant_info_for_generator<'tcx>(
             }
         })
         .collect();
+
+    // The first three variants are hardcoded to be `UNRESUMED`, `RETURNED` and `POISONED`.
+    // We will move the `RETURNED` and `POISONED` elements to the end so we
+    // are left with a sorting order according to the generators yield points:
+    // First `Unresumed`, then the `SuspendN` followed by `Returned` and `Panicked` (POISONED).
+    let end_states = variant_infos.drain(1..=2);
```

Comment:
I know this is a draft, but noting this just so I don't forget -- reordering these so that they don't line up with the generator variants in https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/sty/struct.GeneratorSubsts.html# is somewhat concerning.

Reply:
Oh wait, duh, I'll move this to the other PR.